R/render.R

```diff
@@ -24,7 +24,11 @@ render <- function(input,
                    envir = parent.frame(),
                    run_pandoc = TRUE,
                    quiet = FALSE,
-                   encoding = getOption("encoding")) {
+                   encoding = getOption("encoding"),
+                   on_exit = NULL) {
```

Comment:
I was thinking that all of this logic could live inside the output format (i.e. the format itself would define an on_exit function in the same way it currently defines e.g. pre_knit and post_knit).